### DISCUSSION TITLE
Question about Passing Strings in a Specific Language to NVDA

### DISCUSSION THREAD

**Discussion by ghost** [nvaccess/nvda#17489] [Q&A] [answered]
Hi, I am wondering if it is possible when I make NVDA speak directly if it is possible to set languages halfway through the string, or if I have to build a buffer where I set the language for each buffer. 

The project is using nvdaControllerClient64.dll, which I am unfamiliar with the documentation on. It just runs a speech method with a string for the purposes of speaking. 

I know in HTML you can use the "lang" attribute to set a language between tags, but I am wondering how I could pass a string that can read a string in a particular language. This is for a video game, so HTML isn't really being used.

For example, the sentence "The French word for Lovebirds, 'Les Inséparables', is very cute" has words in the middle that should be pronounced in French.

Do I have to buffer it? so one string is in English, the next in French, then go back to English in this example? How can I set the language chosen for speech before executing the speech command? Or is it possible to use some kind of tag to label the French text?

For context, this is for a customizable flashcard game, so its important for it to support multiple language synthesizers.

**Comment by hwf1324** [comment]
CC: @gexgd0419

**Comment by CyrilleB79** [comment]
Hi @badgernested 

I guess you have already tried NVDA's examples?
I have tried to adapt https://github.com/nvaccess/nvda/blob/master/extras/controllerClient/examples/example_python.py with a `lanng` tag, but without success. However, I have almost no experience and SSML, so you may be more lucky than I.

**Reply by ghost** [reply, marked as answer]
Hey there @CyrilleB79 ,

I got the script to work when using the most recent version of NVDA. Per the documentation:

> Version 2.0 of the controller client was introduced in NVDA 2024.1, offering the following additional functions compared to version 1.0:
> - nvdaController_getProcessId
> - nvdaController_speakSsml
